docs/source/reference-core.rst

```diff
@@ -834,13 +834,14 @@ Nursery objects provide the following interface:
 
       Note that this is *not* an async function and you don't use await
       when calling it. It sets up the new task, but then returns
-      immediately, *before* it has a chance to run. The new task won’t
-      actually get a chance to do anything until some later point when
-      you execute a checkpoint and the scheduler decides to run it.
-      If you want to run a function and immediately wait for its result,
+      immediately, *before* it has a chance to run. At the next
+      checkpoint, (which might be closing of the nursery context), all
+      tasks requested with :meth:`start_soon` will begin in no
+      particular order. If you want to run a function to completion,
       then you don't need a nursery; just use ``await async_fn(*args)``.
-      If you want to wait for the task to initialize itself before 
-      continuing, see :meth:`start()`.
+      If you want to ensure that the child task does something before
```

Comment:
What do you find to be clear from "ensure that the child task does something before the parent moves onto the next line" that isn't clear from "wait for the task to initialize itself before continuing"? I don't really see a substantive difference, and tend to prefer the shorter/punchier version, but I might be missing something. (I think being clear about "the *child* task" and maybe "before continuing *execution of the parent*" are worthwhile changes, but I'm sketchy on the more complex grammatical structure.)

Reply:
Initialize sounds technical, and implies to me that there's some sort of __init__ function being called, or that there's something other than a function being called. Rewrote to the punchier "If you want to run some part of the child task before the parent task proceeds, see :meth:`start()`."